modules/ipc/src/zenoh/dynamic_subscriber.cpp

```diff
@@ -72,25 +72,23 @@ void DynamicSubscriber::onPublisher(const EndpointInfo& info) {
 }
 
 void DynamicSubscriber::onPublisherAdded(const EndpointInfo& info) {
-  std::optional<serdes::TypeInfo> optional_type_info;
   if (subscribers_.contains(info.topic)) {
     heph::log(heph::ERROR, "trying to add subscriber for topic but one already exists", "topic", info.topic);
     return;
   }
 
+  auto type_info = topic_db_->getTypeInfo(info.topic);
```

Comment:
I have some changes for this part of the code in my branch as well + in the topic database. If the type retrieval fails, I was just not finishing the onPublisherAdded callback.

Reply:
I think it is easier if you make the PR for that